### PR TITLE
chore: update consentRequests simulator resp to support all error cases

### PR DESCRIPTION
**Pull request by sridharvoruganti**

### CODE REVIEW

**Review by lewisdaly — APPROVED** [review]

File: test/unit/shared/util.test.ts [hunk 1]
```diff
@@ -89,3 +90,15 @@ describe('shared/reformatError', (): void => {
   })
 
 })
+describe('shared/throwMojaloopFSPIOPError', (): void => {
```
Comment (eoln) [inline test/unit/shared/util.test.ts:93]: there is a need to refactor test for mkMojaloopFSPIOPError

File: src/models/inbound/dfspConsentRequests.model.ts [hunk 2]
```diff
@@ -34,11 +34,12 @@ import {
   thirdparty as tpAPI
 } from '@mojaloop/api-snippets'
 import inspect from '~/shared/inspect'
-import { reformatError } from '~/shared/util'
+import { reformatError, throwMojaloopFSPIOPError } from '~/shared/util'
```
Comment (eoln) [inline src/models/inbound/dfspConsentRequests.model.ts:37]:
```suggestion
import { reformatError, mkMojaloopFSPIOPError } from '~/shared/util'
```

File: src/models/inbound/dfspConsentRequests.model.ts [hunk 3]
```diff
@@ -85,14 +86,15 @@ export class DFSPConsentRequestsModel
     const { request, toParticipantId } = this.data
 
     try {
-      const response = await this.dfspBackendRequests.validateConsentRequests(request)
+      const response =
+        await this.dfspBackendRequests.validateConsentRequests(request) as BackendValidateConsentRequestsResponse
 
       if (!response) {
-        throw new Errors.MojaloopFSPIOPError('', '', '', Errors.MojaloopApiErrorCodes.TP_FSP_CONSENT_SCOPES_ERROR)
+        throwMojaloopFSPIOPError(Errors.MojaloopApiErrorCodes.TP_CONSENT_REQ_VALIDATION_ERROR)
```
Comment (eoln) [inline src/models/inbound/dfspConsentRequests.model.ts:93]: when we use `throw` here not from function, it helps TS compiler & linter a lot.
```suggestion
        throw mkMojaloopFSPIOPError(Errors.MojaloopApiErrorCodes.TP_CONSENT_REQ_VALIDATION_ERROR)
```

**Review by eoln — CHANGES_REQUESTED** [review]

File: src/models/inbound/dfspConsentRequests.interface.ts [hunk 4]
```diff
@@ -40,8 +41,11 @@ import * as OutboundAPI from '~/interface/outbound/api_interfaces'
 
 export interface BackendValidateConsentRequestsResponse {
   isValid: boolean
-  authChannels: OutboundAPI.Schemas.AuthChannels
-  authUri?: string
+  data: {
```
[the remark this answers is not in the document]
Comment (sridharvoruganti) [reply src/models/inbound/dfspConsentRequests.interface.ts:44]: I think data is fine, `isValid`  is validation response and data contains some extra information returned by dfsp

File: src/shared/util.ts [hunk 5]
```diff
@@ -59,6 +59,16 @@ async function reformatError (err: Error): Promise<Errors.MojaloopApiErrorObject
   return new Errors.MojaloopFSPIOPError(err, '', '', mojaloopErrorCode).toApiErrorObject()
 }
 
+/**
+ * @function throwMojaloopFSPIOPError
+ * @description Helper function to throw MojaloopFSPIOPError
+ * @param {MojaloopApiErrorCode} errorCode
+ */
+function throwMojaloopFSPIOPError (errorCode: Errors.MojaloopApiErrorCode): void {
+  throw new Errors.MojaloopFSPIOPError('', '', '', errorCode)
```
[the remark this answers is not in the document]
Comment (sridharvoruganti) [reply src/shared/util.ts]: Ok, thanks for the detailed suggestion

File: src/models/inbound/dfspConsentRequests.model.ts [hunk 6]
```diff
@@ -85,14 +86,15 @@ export class DFSPConsentRequestsModel
     const { request, toParticipantId } = this.data
 
     try {
-      const response = await this.dfspBackendRequests.validateConsentRequests(request)
+      const response =
+        await this.dfspBackendRequests.validateConsentRequests(request) as BackendValidateConsentRequestsResponse
```
[the remark this answers is not in the document]
Comment (sridharvoruganti) [reply src/models/inbound/dfspConsentRequests.model.ts]: ok removed it

File: test/unit/shared/util.test.ts (hunk 1 above)
Comment (sridharvoruganti) [reply test/unit/shared/util.test.ts]: Ok done

**Review by eoln — APPROVED** [review]

**Pull request merged**